### PR TITLE
Update openshift-cluster-api-provider-agent-master to reference assisted-service-index container images in quay instead of the internal registry of Prow

### PR DESCRIPTION
**Pull request by eranco74**
cc: @osherdp @adriengentil

### CODE REVIEW

File: ci-operator/config/openshift/cluster-api-provider-agent/openshift-cluster-api-provider-agent-master.yaml [hunk 1]
```diff
@@ -19,10 +19,10 @@ base_images:
     name: ocm-2.7
     namespace: edge-infrastructure
     tag: assisted-service
-  assisted-service-operator-index:
-    name: ocm-2.7
+  assisted-service-index:
+    name: assisted-service-index
```
Comment (osherdp) [inline ci-operator/config/openshift/cluster-api-provider-agent/openshift-cluster-api-provider-agent-master.yaml:23]: I think you need to change below:
```
- as: e2e-ai-operator-ztp-capi
  optional: true
  steps:
    cluster_profile: packet-assisted
    dependencies:
      ...
      INDEX_IMAGE: assisted-service-index
      ...
```

**Review by osherdp — CHANGES_REQUESTED** [review]

**Comment by osherdp** [comment]
/hold
until seeing it all passing

**Comment by adriengentil** [comment]
Just in case it fails because the bundle is not available, I'm fixing it here: https://github.com/openshift/release/pull/33246

**Comment by osherdp** [comment]
failed on dev-scripts setup I think
/test pj-rehearse

**Comment by eranco74** [comment]
failed on dev-scripts setup again
/test pj-rehearse

**Comment by eranco74** [comment]
Same change is required in ocm-2.6
https://github.com/openshift/release/pull/33293

**Pull request merged**